Utils/src/L1NonPrefiringProbProducer.cc

```diff
@@ -0,0 +1,348 @@
+  std::array<double,3> getNonPrefiringRateMuon( double eta, double phi, double pt) const;
+ 
+  edm::EDGetTokenT<std::vector< pat::Photon> > photons_token_; 
+  edm::EDGetTokenT<std::vector< pat::Jet> > jets_token_;
+  edm::EDGetTokenT<std::vector< pat::Muon> > muons_token_;
+
+  TH2F* h_prefmap_photon; 
+  TH2F* h_prefmap_jet;
+  TF1* h_prefparam_muon_0p00to0p20;
+  TF1* h_prefparam_muon_0p20to0p30;
+  TF1* h_prefparam_muon_0p30to0p55;
+  TF1* h_prefparam_muon_0p55to0p83;
+  TF1* h_prefparam_muon_0p83to1p24;
+  TF1* h_prefparam_muon_1p24to1p40;
+  TF1* h_prefparam_muon_1p40to1p60;
+  TF1* h_prefparam_muon_1p60to1p80;
+  TF1* h_prefparam_muon_1p80to2p10;
+  TF1* h_prefparam_muon_2p10to2p25;
+  TF1* h_prefparam_muon_2p25to2p40;
+  TF1* h_prefparam_muon_hotspot;
+  std::string dataeraecal_;
+  std::string dataeramuon_;
+  bool useEMpt_;
+  double prefiringRateSystUncECAL_;
+  double prefiringRateSystUncMuon_;
+  double jet_max_muon_fraction_;
+
+};
+
+L1NonPrefiringProbProducer::L1NonPrefiringProbProducer(const edm::ParameterSet& iConfig) :
+  photons_token_(consumes<std::vector<pat::Photon>>(iConfig.getParameter<edm::InputTag>("ThePhotons"))),
+  jets_token_(consumes<std::vector<pat::Jet>>(iConfig.getParameter<edm::InputTag>("TheJets"))),
+  muons_token_(consumes<std::vector<pat::Muon>>(iConfig.getParameter<edm::InputTag>("TheMuons"))),
+  dataeraecal_(iConfig.getParameter<std::string>("DataEraECAL")),
+  dataeramuon_(iConfig.getParameter<std::string>("DataEraMuon")),
+  useEMpt_(iConfig.getParameter<bool>("UseJetEMPt")),
+  prefiringRateSystUncECAL_(iConfig.getParameter<double>("PrefiringRateSystematicUnctyECAL")),
+  prefiringRateSystUncMuon_(iConfig.getParameter<double>("PrefiringRateSystematicUnctyMuon")),
+  jet_max_muon_fraction_(iConfig.getParameter<double>("JetMaxMuonFraction"))
+{
+  std::string fnameecal =  iConfig.getParameter<std::string>("L1MapsECAL");
+  TFile* file_ecalprefiringmaps = TFile::Open(fnameecal.c_str(),"read");
+
+  std::string mapphotonfullname = "L1prefiring_photonptvseta_"+ dataeraecal_; 
+  file_ecalprefiringmaps->ls();
+  h_prefmap_photon = (TH2F*)file_ecalprefiringmaps->Get(mapphotonfullname.c_str());
+  h_prefmap_photon->SetDirectory(0);
+
+  std::string mapjetfullname = "L1prefiring_jet";
+  mapjetfullname += (useEMpt_ ? "empt" : "pt");
+  mapjetfullname += "vseta_" + dataeraecal_;
+  h_prefmap_jet = (TH2F*)file_ecalprefiringmaps->Get(mapjetfullname.c_str());
+  h_prefmap_jet->SetDirectory(0);
+  file_ecalprefiringmaps->Close();
+
+  std::string fnamemuon =  iConfig.getParameter<std::string>("L1ParamsMuon");
+  TFile* file_muonprefiringparams = TFile::Open(fnamemuon.c_str(), "read");
+
+  std::string mapmuonfullname = "L1prefiring_muonparam_HotSpot_" + dataeramuon_;
+  h_prefparam_muon_hotspot = (TF1*)file_muonprefiringparams->Get(mapmuonfullname.c_str());
+
+  mapmuonfullname = "L1prefiring_muonparam_0.0To0.2_" + dataeramuon_;
```

Comment:
to reduce duplication, make `std::vector<std::pair<double,std::string>>` and then loop over it to fill the map suggested in the previous comment.

Reply:
Compactified in e9088f6